src/components/RangeInput.vue

```diff
@@ -0,0 +1,270 @@
+<template>
+  <div :class="bem()">
+    <slot name="header" />
+
+    <form @submit.prevent="onSubmit(refinement)">
+      <input
+        type="number"
+        :class="bem('input', 'from')"
+        :min="range.min"
+        :max="range.max"
+        :step="step"
+        :placeholder="rangeForRendering.min"
+        :value="refinementForRendering.min"
+        @input="refinement.min = $event.target.value"
+      />
+
+      <slot name="separator">
+        <span :class="bem('separator')">
+          to
+        </span>
+      </slot>
+
+      <input
+        type="number"
+        :class="bem('input', 'to')"
+        :min="range.min"
+        :max="range.max"
+        :step="step"
+        :placeholder="rangeForRendering.max"
+        :value="refinementForRendering.max"
+        @input="refinement.max = $event.target.value"
+      />
+      <slot name="submit">
+        <button :class="bem('submit')">Ok</button>
+      </slot>
+    </form>
+
+    <slot name="footer" />
+  </div>
+</template>
+
+<script>
+import algoliaComponent from '../component';
+import { FACET_OR } from '../store';
+
+export default {
+  mixins: [algoliaComponent],
+  props: {
+    attributeName: {
+      type: String,
+      required: true,
+    },
+    min: {
+      type: Number,
+    },
+    max: {
+      type: Number,
+    },
+    defaultRefinement: {
+      type: Object,
+      default() {
+        return {};
+      },
+    },
+    precision: {
+      type: Number,
+      default: 0,
+      validator(value) {
+        return value >= 0;
+      },
+    },
+  },
+
+  data() {
+    return {
+      blockClassName: 'ais-range-input',
+    };
+  },
+
+  created() {
+    const { min: minValue, max: maxValue } = this.defaultRefinement;
+
+    let min;
+    if (minValue !== undefined) {
+      min = minValue;
+    } else if (this.min !== undefined) {
+      min = this.min;
+    }
+
+    let max;
+    if (maxValue !== undefined) {
+      max = maxValue;
+    } else if (this.max !== undefined) {
+      max = this.max;
+    }
```

Comment:
could be rewritten as a ternary without reassignment, but no big deal 😄

Reply:
How? I'm really interested in your proposal 🙂 

I **really** don't like reassignment, I usually use nested ternary but it's forbidden by ESLint (for good reasons I think). Note that I need to keep the value `undefined` when neither the `defaultValue` nor the `max` property is passed.

Reply:
not sure if it's better, but an option could be `const max = maxValue || this.max`, granted that it will take more than `undefined` of `maxValue`

Reply:
Yep, but if the value passed is `0` it will consider falsy and then fallback on the other value (this case is only relevant for `min`).